server/src/pages/homepage/controller.js

```diff
@@ -5,7 +5,22 @@ let model;
 
 exports.launch = function(m) {
   model = m;
-  render();
+  if (window.wantsauth) {
+    if (window.wantsauth.getAuthData()) {
+      Object.assign(model, window.wantsauth.getAuthData());
+    } else {
+      window.wantsauth.addAuthDataListener((data) => {
+        Object.assign(model, data);
+        render();
+      });
+    }
+  }
+  try {
+    render();
+  } catch (e) {
+    window.Raven.captureException(e);
+    throw e;
+  }
```

Comment:
Why is this `render()` call in a `try-catch` but the one above in the event callback is not?

Reply:
Followed pattern from shot page, for homepage it could be overkill